src/Microsoft.Diagnostics.EventFlow.Inputs.Log4net/Log4netInput.cs

```diff
@@ -0,0 +1,135 @@
+
+using log4net.Core;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using log4net;
+using log4net.Appender;
+using log4net.Repository.Hierarchy;
+using Microsoft.Diagnostics.EventFlow.Configuration;
+using Validation;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs
+{
+    public class Log4NetInput : AppenderSkeleton, IObservable<EventData>, IDisposable
+    {
+        private static readonly IDictionary<Level, LogLevel> ToLogLevel =
+            new Dictionary<Level, LogLevel>
+            {
+                [Level.Verbose] = LogLevel.Verbose,
+                [Level.Debug] = LogLevel.Verbose,
+                [Level.Info] = LogLevel.Informational,
+                [Level.Warn] = LogLevel.Warning,
+                [Level.Error] = LogLevel.Error,
+                [Level.Fatal] = LogLevel.Critical
+            };
+
+        private IHealthReporter healthReporter;
+        private IConfiguration configuration;
+        private EventFlowSubject<EventData> subject;
+        private Log4netConfiguration _log4NetInputConfiguration;
+        private Hierarchy eventFlowRepo;
+
+        public Log4NetInput(IConfiguration configuration, IHealthReporter healthReporter)
+        {
+            Requires.NotNull(healthReporter, nameof(healthReporter));
+            Requires.NotNull(configuration, nameof(configuration));
+
+            var log4NetInputConfiguration = new Log4netConfiguration();
+            try
+            {
+                configuration.Bind(log4NetInputConfiguration);
+            }
+            catch
+            {
+                healthReporter.ReportProblem($"Invalid {nameof(log4NetInputConfiguration)} configuration encountered: '{configuration}'",
+                    EventFlowContextIdentifiers.Configuration);
+                throw;
+            }
+
+            Initialize(log4NetInputConfiguration, healthReporter);
+        }
+
+        public Log4NetInput(Log4netConfiguration log4NetConfigs, IHealthReporter healthReporter)
+        {
+            Requires.NotNull(log4NetConfigs, nameof(log4NetConfigs));
+            Requires.NotNull(healthReporter, nameof(healthReporter));
+
+            Initialize(log4NetConfigs, healthReporter);
+        }
+
+        private void Initialize(Log4netConfiguration myLog4NetConfig, IHealthReporter myHealthReporter)
+        {
+            this.healthReporter = myHealthReporter;
+            this._log4NetInputConfiguration = myLog4NetConfig;
+            this.subject = new EventFlowSubject<EventData>();
+
+            //Can we determine if the repo exists without try/catch
+            try
+            {
+                eventFlowRepo = (Hierarchy)LogManager.CreateRepository("EventFlow");
+            }
+            catch (LogException ex)
+            {
+                eventFlowRepo = (Hierarchy) LogManager.GetRepository("EventFlow");
+            }
+            eventFlowRepo.Root.AddAppender(this);
```

Comment:
I do not know much about Log4Net API--if the repository exists (and `Configured` property was set to true), is it going to be a problem to call `AddAppender()` again?